src/pages/index.js

```diff
@@ -93,22 +203,72 @@ const editProfilePopup = new PopupWithForm({
 });
 editProfilePopup.setEventListeners();
 
+profileEditBtn.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileInputTitle.value = userData.title;
+  profileInputDescription.value = userData.description;
+  profileEditFormValidator.resetValidation();
+  editProfilePopup.open();
+});
+
+// AVATAR MODAL AND METHODS -----
+
+const handleAvatarFormSubmit = (data) => {
+  function makeRequest() {
+    return api.updateProfileAvatar(data.link).then(() => {
+      userInfo.changeAvatar(data.link);
+      editAvatarPopup.close();
+    });
+  }
+  handleSubmit(makeRequest, editAvatarPopup);
+};
+
+const editAvatarPopup = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: handleAvatarFormSubmit,
+});
+editAvatarPopup.setEventListeners();
+
+editAvatarButton.addEventListener("click", () => {
+  avatarUrlInput.value = userInfo.getUserInfo().avatar;
+  avatarFormValidator.resetValidation();
+  editAvatarPopup.open();
+});
+
+// ADD CARD MODAL AND METHODS -----
+
 const addCardPopup = new PopupWithForm({
   popupSelector: "#card-add-modal",
   handleFormSubmit: handleAddCardFormSubmit,
 });
 addCardPopup.setEventListeners();
 
+function handleAddCardFormSubmit(data) {
+  function makeRequest() {
+    return api
+      .addCard({ name: data.name, link: data.link })
+      .then((cardData) => {
+        section.addItem(createCard(cardData));
+        cardAddForm.reset();
+        addCardPopup.close();
+        addCardFormValidator.disableButton();
+      });
+  }
+  handleSubmit(makeRequest, addCardPopup);
+}
+
 cardAddNewBtn.addEventListener("click", () => {
   addCardPopup.open();
-  addCardFormValidator.toggleButtonState();
-  console.log("hello, world");
+  addCardFormValidator.resetValidation();
```

Comment:
`  addCardFormValidator.resetValidation();
`

Please, delete it

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs.